modules/openy_features/openy_node/modules/openy_node_alert/src/Service/AlertManager.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\openy_node_alert\Service;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+
+/**
+ * Class AlertManager.
+ *
+ * @package Drupal\openy_node_alert\Service
+ */
+class AlertManager {
+
+  /**
+   * The alert builders array.
+   *
+   * @var \Drupal\openy_node_alert\Service\AlertBuilderInterface[]
+   */
+  protected $alerts = [];
+
+  /**
+   * An array with sorted alerts builders by priority, NULL otherwise.
+   *
+   * @var null|array
+   */
+  protected $alertsSorted = NULL;
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $nodeStorage;
+
+  /**
+   * Constructs the Alert manager.
+   *
+   * @param EntityTypeManagerInterface $entity_type_manager
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->nodeStorage = $entity_type_manager->getStorage('node');
+  }
+
+  /**
+   * Adds alerts builders to internal service storage.
+   *
+   * @param \Drupal\openy_node_alert\Service\AlertBuilderInterface $alert
+   *   The alert service.
+   * @param int $priority
+   *   The service priority.
+   */
+  public function addBuilder(AlertBuilderInterface $alert, $priority = 0) {
+    $this->alerts[$priority][] = $alert;
+    // Reset sorted status to be resorted on next call.
+    $this->alertsSorted = NULL;
+  }
+
+  /**
+   * Sorts alert services.
+   *
+   * @return \Drupal\openy_node_alert\Service\AlertBuilderInterface[]
+   *   The sorted messages services.
+   */
+  protected function sortAlerts() {
+    $sorted = [];
+    krsort($this->alerts);
+
+    foreach ($this->alerts as $alerts) {
+      $sorted = array_merge($sorted, $alerts);
+    }
+
+    return $sorted;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(EntityInterface $node) {
+    return TRUE;
+  }
```

Comment:
I believe this method is not needed here - it is a manager, not an alert builder service.